src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphOperations.cs

```diff
@@ -774,6 +816,49 @@ private static CyclesAndDowngrades CreateState(List<GraphNode<RemoteResolveResul
             };
         }
 
+        private static void DetectAndMarkAmbiguousCentralTransitiveDependencies<TItem>(this GraphNode<TItem> root, Tracker<TItem> tracker)
+        {
+            // if a central transitive node has all parents disputed or ambiguous mark it and its children ambiguous
+            root.InnerNodes
+                .Where(x => x.Item.IsCentralTransitive && x.Disposition == Disposition.Acceptable)
+                .ForEach(n =>
+                {
+                    if (!n.ParentNodes.Any(p => !(tracker.IsDisputed(p.Item) || tracker.IsAmbiguous(p.Item))))
+                    {
+                        tracker.MarkAmbiguous(n.Item);
+                        n.ForEach(x => tracker.MarkAmbiguous(x.Item));
+                    }
+                });
+        }
+
+        private static void RejectCentralTransitiveBecauseOfRejectedParents<TItem>(this GraphNode<TItem> root)
```

Comment:
This looks really expensive to run on each node both from a big O perspective and number of allocations perspective. That's why I was asking about performance tests. The ASP.NET Core repo has > 100 central dependencies, does this perform well at that scale?

Reply:
Yes the big O can be large depending on the graph structure. I did run benchmarks for rejects/acceptances scenarios. The results are below. I will still keep thinking of solutions to reduce it. 
The total central dependencies count is important, for this scenario more important is the count of the subset that becomes transitive dependencies. 

|Name|Time milliseconds|
|---|---|
2_10_2_AnalyseGraphWith4AcceptedCentralTransitiveDeps | 2
2_10_2_AnalyseGraphWith4RejectedCentralTransitiveDependencies | 3.56
2_10_2_AnalyseGraphWithoutCentralTransitiveDepsAsync_HalfGraphRejected | 2.66
2_10_4_AnalyseGraphWith16AcceptedCentralTransitiveDeps | 2.61
2_10_4_AnalyseGraphWith16RejectedCentralTransitiveDependencies | 4.06
2_10_4_AnalyseGraphWithoutCentralTransitiveDepsAsync_HalfGraphRejected | 2.82
|2_10_6_AnalyseGraphWith64AcceptedCentralTransitiveDeps|3.09|
|2_10_6_AnalyseGraphWith64RejectedCentralTransitiveDependencies|4.31|
|2_10_6_AnalyseGraphWithoutCentralTransitiveDepsAsync_HalfGraphRejected|3.09|